lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,427 @@
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toHaveClass('nx-combobox__menu');
+  });
+
+  it('sets the listbox role on the dropdown', function() {
+    const { getByRole } = rtlRender({ matches: [{ id: '1', displayName: 'Foo' }] }),
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toHaveAttribute('role', 'listbox');
+  });
+
+  it('sets an id on the dropdown and references it in the combobox input aria-controls', function() {
+    const { getByRole } = rtlRender({ matches: [{ id: '1', displayName: 'Foo' }] }),
+        inputElement = getByRole('combobox'),
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toHaveAttribute('id');
+    expect(inputElement).toHaveAttribute('aria-controls', dropdownElement.id);
+  });
+
+  it('renders an alert dropdown with nx_combobox__alert class', function() {
+    const { container } = rtlRender(),
+        alertDropdownElement = container.querySelector('.nx-combobox__alert');
+
+    expect(alertDropdownElement).toBeTruthy();
+    expect(alertDropdownElement).toHaveClass('nx-combobox__alert');
+  });
+
+  it('sets the alert role on the alert dropdown when it is in loading, error, or empty states', function() {
+    expect(rtlRender({ matches: [] }).container.querySelector('.nx-combobox__alert')).toHaveAttribute('role', 'alert');
+    expect(rtlRender({ loading: true }).container.querySelector('.nx-combobox__alert'))
+        .toHaveAttribute('role', 'alert');
+    expect(rtlRender({ loadError: 'boo' }).container.querySelector('.nx-combobox__alert'))
+        .toHaveAttribute('role', 'alert');
+  });
+
+  it('sets an id on the alert dropdown and references it in the combobox input aria-describedby', function() {
+    const { getByRole, container } = rtlRender(),
+        inputElement = getByRole('combobox'),
+        alertDropdownElement = container.querySelector('.nx-combobox__alert')!;
+
+    expect(alertDropdownElement).toHaveAttribute('id');
+    expect(inputElement).toHaveAttribute('aria-describedby', alertDropdownElement.id);
+  });
+
+  it('sets aria-live on the alert dropdown to "polite"', function() {
+    expect(rtlRender().container.querySelector('.nx-combobox__alert')).toHaveAttribute('aria-live', 'polite');
+  });
+
+  it('sets aria-busy on the alert dropdown if loading is true', function() {
+    expect(rtlRender().container.querySelector('.nx-combobox__alert')).toHaveAttribute('aria-busy', 'false');
+    expect(rtlRender({ loading: true }).container.querySelector('.nx-combobox__alert'))
+        .toHaveAttribute('aria-busy', 'true');
+  });
+
+  it('renders a loading spinner when the `loading` prop is true', function() {
+    const { container } = rtlRender({ loading: true }),
+        loadingElement = container.querySelector('.nx-loading-spinner');
+    expect(loadingElement).toBeTruthy();
+  });
```

Comment:
Covered by visual test; the classname is an implementation detail

Reply:
Removed, updated here 4029d58f1778f46cf53cdedad783b595e4a9da4e.